File: netapp-manageability-sdk-9.3/src/sample/Data_ONTAP/Ruby/optmgmt.rb
#============================================================#
#                                                            #
# $ID$                                                       #
#                                                            #
# optmgmt.rb                                                 #
#                                                            #
# Sample code for the following APIs:                        #
#               options-get                                  #
#               options-set                                  #
#               options-list-info                            #
#                                                            #
# Copyright 2011 Network Appliance, Inc. All rights          #
# reserved. Specifications subject to change without notice. #
#                                                            #
# This SDK sample code is provided AS IS, with no support or #
# warranties of any kind, including but not limited to       #
# warranties of merchantability or fitness of any kind,      #
# expressed or implied.  This code is subject to the license #
# agreement that accompanies the SDK.                        #
#                                                            #
#============================================================#

$:.unshift '../../../../lib/ruby/NetApp'
require 'NaServer'

def print_usage
    print ("optmgmt.rb <storage> <user> <password> <operation> [<value1>] ")
    print ("[<value2>] \n")
    print ("<storage> -- Storage system name\n")
    print ("<user> -- User name\n")
    print ("<password> -- Password\n")
    print ("<operation> -- Operation to be performed: get/set/optionsList\n")
    print ("[<value1>] -- Name of the option\n")
    print ("[<value2>] -- Value to be set\n")
    exit 
end


$args = ARGV.length
if($args < 4)
    print_usage()
end
$storage = ARGV[0]
$user = ARGV[1]
$pw = ARGV[2]
$option = ARGV[3]
$value1 = nil
$value2 = nil

if($args > 5)
    $value1 = ARGV[4]
    $value2 = ARGV[5]	
elsif($args == 5)
    $value1 = ARGV[4]
end


def get_option_info()
    if($args < 5)
        print_usage
    end
    out = $s.invoke("options-get", "name", $value1)
    if(out.results_status == "failed")
        print(out.results_reason() + "\n")
        exit
    end	
    print("-------------------------------------------------------------\n")
    print("Cluster constraint: " + out.child_get_string("cluster-constraint"))
    print("\n")
    print("Value: " + out.child_get_string("value") + "\n")
    print("-------------------------------------------------------------\n")
end


def set_option_info()
    if($args < 6)
        print_usage
    end
    out = $s.invoke("options-set", "name", $value1, "value", $value2)
    if(out.results_status() == "failed")
        print(out.results_reason() + "\n")
        exit
    end	
    print ("-------------------------------------------------------------\n")
    print("Cluster constraint: "+out.child_get_string("cluster-constraint"))
    print("\n")
    if(out.child_get_string("message"))
        print("Message: " + out.child_get_string("message") + "\n")
        print("-------------------------------------------------------------\n")
    end
end


def options_list_info()
    out = $s.invoke("options-list-info")
    if(out.results_status() == "failed")
        print(out.results_reason() + "\n")
        exit
    end
    options_info = out.child_get("options")
    result = options_info.children_get()
    result.each do |opt|
	print("------------------------------------------------------------\n")
        print("Cluster constraint: ")
        print(opt.child_get_string("cluster-constraint") + "\n")
        print("Name: " + opt.child_get_string("name") + "\n")
        print("Value: " + opt.child_get_string("value") + "\n")
    end
end


def main
    $s = NaServer.new($storage, 1, 3)
    $s.set_admin_user($user, $pw)
    if($option == "get")
        get_option_info()
    elsif($option == "set")
        set_option_info()
    elsif($option == "optionsList")
        options_list_info()
    else
        print("Invalid Option \n")
        print_usage()
    end
end
	
main()

File: netapp-manageability-sdk-9.3/src/sample/Data_ONTAP/DotNet/VB.NET/optmgmt/optmgmt.vb
'==========================================================================='
'                                                                           '
'                                                                           '
' optmgmt.vb                                                                '
'                                                                           '
' ONTAPI API lists option information, get value of a specific              '
' option, and set value for a specific option.		                        '
'                                                                           '
' Copyright 2008 NetApp, Inc. All rights                                    '
' reserved. Specifications subject to change without notice.                '
'                                                                           '
' This SDK sample code is provided AS IS, with no support or                '
' warranties of any kind, including but not limited to                      '
' warranties of merchantability or fitness of any kind,                     '
' expressed or implied.  This code is subject to the license                '
' agreement that accompanies the SDK.                                       '
'                                                                           '
'                                                                           '
' Usage:							                                        '
' optmgmt <filer> <user> <password> [<operation>] [<optionName>] [<value>]  '
' <filer> 	-- Name/IP address of the filer			                        '
' <user>  	-- User name					                                '
' <password> 	-- Password					                                '
' <operation> 	-- get/set					                                '
' <optionName>	-- Name of the option on which get/set operation            '
'		   needs to be performed			                                '
' <value> 	-- This is required only for set operation. 	                '
'		   Provide the value that needs to be assigned for                  '
'		   the option					                                    '
'                                                                           '
'==========================================================================='

Imports System
Imports System.Collections.Generic
Imports NetApp.Manage

Module OptMgmt
    Public Sub Usage()
        Console.WriteLine(Environment.NewLine & "Usage: optmgmt <filer> <username> <passwd> [<operation(get/set)> [<optionName>] [<value>]]")
    End Sub
    Public Sub Main()
        Dim args As String() = Environment.GetCommandLineArgs()

        If (args.Length < 4) Then
            Usage()
            Exit Sub
        End If

        Dim serverName As String = args(1)
        Dim user As String = args(2)
        Dim passwd As String = args(3)
        Dim operation As String = ""
        Dim name As String = ""
        Dim value As String = ""

        
        Try
            'Initialize connection to server, and request version 1.0 of the API set
            Dim server As New NaServer(serverName, 1, 0)
            Dim output As NaElement = Nothing

            server.ServerType = NaServer.SERVER_TYPE.FILER
            server.TransportType = NaServer.TRANSPORT_TYPE.HTTP
            server.SetAdminUser(user, passwd)

            If args.Length > 4 Then
                operation = args(4)
                'Find the given operation
                If (operation = "get") Then
                    If (args.Length < 6) Then
                        Console.WriteLine("operation get requires an option name")
                        Exit Sub
                    End If
                    name = args(5)
                                        
                    'Invoke option get Info ONTAPI API
                    output = server.Invoke("options-get","name",name)
                    Console.WriteLine("----------------------------")
                    Console.WriteLine("Option Value:" & output.GetChildContent("value"))
                    Console.WriteLine("Cluster Constraint:" & output.GetChildContent("cluster-constraint"))
                    Console.WriteLine("----------------------------")
                Else
                    If (operation = "set") Then
                        If (args.Length < 7) Then
                            Console.WriteLine("operation set requires option name and value pair")
                            Exit Sub
                        End If
                        name = args(5)
                        value = args(6)
                        output = server.Invoke("options-set","name",name,"value",value)
                        Console.WriteLine("----------------------------")
                        If (output.GetChildContent("message") <> Nothing) Then
                            Console.WriteLine("Message:" & output.GetChildContent("message"))
                        End If
                        Console.WriteLine("Cluster Constraint:" & output.GetChildContent("cluster-constraint"))
                        Console.WriteLine("----------------------------")
                    Else
                        Console.WriteLine("Invalid Operation")
                        Usage()
                        Exit Sub
                    End If
                    End If
            Else
                
                output = server.Invoke("options-list-info")
                Dim optList As List(Of NaElement) = output.GetChildByName("options").GetChildren()
                Dim total As Integer = optList.Count
                Dim optInfo As NaElement
                Dim index As Integer

                For index = 0 To total - 1
                    Console.WriteLine("---------------------------------")
                    optInfo = optList(index)
                    Console.WriteLine("Option Name:" & optInfo.GetChildContent("name"))
                    Console.WriteLine("Option Value:" & optInfo.GetChildContent("value"))
                    Console.WriteLine("Cluster Constraint:" & optInfo.GetChildContent("cluster-constraint"))
                Next index
            End If
        Catch ex As Exception
            Console.WriteLine(ex.Message)
        End Try
    End Sub
End Module